### ISSUE TITLE
[F-B-3] Training Pipeline - ML Model Training & Validation

### ISSUE THREAD

**Issue by siven76** [siven76/PA-AI#12]
## Feature Description
Implement a config-driven training pipeline using Hydra that performs rolling walk-forward validation with train→val→test windows, storing metrics to MLflow for model tracking and comparison.

## Acceptance Criteria
- [ ] Config-driven pipeline using Hydra configuration management
- [ ] Rolling walk-forward validation methodology
- [ ] Train→Validation→Test window splitting
- [ ] MLflow integration for experiment tracking
- [ ] Support for multiple model architectures (TCN, XGBoost)
- [ ] Automated hyperparameter optimization
- [ ] Model versioning and artifact storage
- [ ] Performance metrics calculation and logging

## Technical Requirements
- **Configuration**: Hydra for hierarchical config management
- **Experiment Tracking**: MLflow for metrics, parameters, and artifacts
- **Validation**: Time-series aware walk-forward splits
- **Models**: Temporal Convolutional Networks, Gradient Boosting
- **Data**: Integration with F-B-2 labeled datasets
- **Storage**: Model artifacts in MLflow model registry

## Pipeline Architecture
```
Config → Data Loading → Feature Engineering → Model Training → Validation → MLflow Logging
  ↓           ↓              ↓                 ↓              ↓           ↓
Hydra     F-B-2 Labels   Technical Ind.    TCN/XGBoost    Walk-Forward  Metrics
YAML      Parquet/CSV    Pattern Features   Training       Time Splits   Artifacts
```

## Walk-Forward Validation
```python
# Example: 12 months data, 3-month windows
Train Window:    [Jan-Mar] → Model A
Validation:      [Apr]     → Tune A
Test:           [May]     → Evaluate A

Train Window:    [Feb-Apr] → Model B  
Validation:      [May]     → Tune B
Test:           [Jun]     → Evaluate B

# Continue rolling forward...
```

## Configuration Structure
```yaml
# config/train.yaml
defaults:
  - model: tcn
  - data: forex_1h
  - optimizer: adam
  - _self_

experiment:
  name: "pa_ai_v1"
  tags: ["price_action", "forex"]
  
data:
  source: "data/labeled/EURUSD_1h.parquet"
  features: ["hammer", "doji", "rsi_14", "sma_20"]
  target: "signal_direction"
  train_months: 3
  val_months: 1
  test_months: 1
  
model:
  architecture: "tcn"
  hidden_size: 64
  num_layers: 4
  kernel_size: 3
  dropout: 0.2
  
training:
  batch_size: 256
  learning_rate: 0.001
  epochs: 100
  early_stopping: 10
  
mlflow:
  tracking_uri: "sqlite:///mlflow.db"
  experiment_name: "pa_ai_training"
```

### Temporal Convolutional Network (TCN)
- **Input**: Sequence of OHLCV + pattern features
- **Architecture**: Dilated convolutions with residual connections
- **Output**: Binary classification (long/short/hold)
- **Advantages**: Parallel training, long sequence modeling

### Gradient Boosting (XGBoost/LightGBM)
- **Input**: Engineered features from recent candles
- **Architecture**: Tree-based ensemble
- **Output**: Binary/multi-class classification
- **Advantages**: Feature importance, fast inference

## Metrics and Evaluation
```python
metrics = {
    "accuracy": accuracy_score(y_true, y_pred),
    "precision": precision_score(y_true, y_pred, average='weighted'),
    "recall": recall_score(y_true, y_pred, average='weighted'),
    "f1": f1_score(y_true, y_pred, average='weighted'),
    "roc_auc": roc_auc_score(y_true, y_proba),
    
    # Trading-specific metrics
    "sharpe_ratio": calculate_sharpe(returns),
    "max_drawdown": calculate_max_drawdown(returns),
    "win_rate": calculate_win_rate(trades),
    "profit_factor": calculate_profit_factor(trades)
}
```

## MLflow Integration
- **Experiments**: Organize by model type and dataset
- **Runs**: Each training run with parameters and metrics
- **Artifacts**: Model files, plots, feature importance
- **Model Registry**: Versioned models with staging/production tags

## Definition of Done
- [ ] Hydra configuration system setup
- [ ] Walk-forward validation implementation
- [ ] MLflow experiment tracking integration
- [ ] TCN and XGBoost model implementations
- [ ] Automated hyperparameter optimization
- [ ] Comprehensive metrics calculation
- [ ] Model artifact storage and versioning
- [ ] Pipeline orchestration and scheduling
- [ ] Unit tests and integration tests
- [ ] Documentation and usage examples
